modules/collect/src/main/java/com/opengamma/collect/timeseries/DenseLocalDateDoubleTimeSeries.java

```diff
@@ -0,0 +1,984 @@
+
+    ArgChecker.isFalse(
+        dates.stream().anyMatch(ld ->
+            ld == null || (dateCalculation == SKIP_WEEKENDS && isWeekend(ld))),
+        "dates must not contain nulls or weekend dates");
+
+    if (dates.isEmpty()) {
+      return EMPTY_SERIES;
+    }
+
+    Pair<LocalDate, LocalDate> extremes = findExtremities(dates);
+
+    LocalDate startDate = extremes.getFirst();
+
+    int size = dateCalculation.calculatePosition(startDate, extremes.getSecond()) + 1;
+    double[] valuesArray = new double[size];
+    if (values.size() != size) {
+      // We have gaps - prefill with NaNs
+      Arrays.fill(valuesArray, Double.NaN);
+    }
+
+    // A utility to zip 2 streams would be useful here!
+
+    Iterator<Double> itValues = values.iterator();
+
+    for (LocalDate date : dates) {
+      valuesArray[dateCalculation.calculatePosition(startDate, date)] = itValues.next();
+    }
+
+    return new DenseLocalDateDoubleTimeSeries(startDate, valuesArray, dateCalculation);
+  }
+
+  private static DenseLocalDateDoubleTimeSeries of(
+      LocalDate startDate,
+      DoubleStream values,
+      DenseTimeSeriesCalculation dateCalculation) {
+    return new DenseLocalDateDoubleTimeSeries(startDate, values.toArray(), dateCalculation);
+  }
+
+  @ImmutableConstructor
+  private DenseLocalDateDoubleTimeSeries(
+      LocalDate startDate,
+      double[] points,
+      DenseTimeSeriesCalculation dateCalculation) {
+
+    this.startDate = ArgChecker.notNull(startDate, "startDate");
+    this.points = ArgChecker.notNull(points, "points").clone();
+    this.dateCalculation = ArgChecker.notNull(dateCalculation, "dateCalculation");
+  }
+
+  private DenseLocalDateDoubleTimeSeries(LocalDate startDate, double[] points) {
+    this(startDate, points, SKIP_WEEKENDS);
+  }
+
+  private static Pair<LocalDate, LocalDate> findExtremities(Collection<LocalDate> dates) {
+    return dates
+        .stream()
+        .reduce(Pair.of(LocalDate.MAX, LocalDate.MIN), (p, ld) -> {
+          LocalDate earliest = p.getFirst();
+          LocalDate latest = p.getSecond();
+          return Pair.of(ld.isBefore(earliest) ? ld : earliest, ld.isAfter(latest) ? ld : latest);
+        }, (p1, p2) -> p1);
+  }
+
+  public static DenseLocalDateDoubleTimeSeries of(LocalDate date, double value) {
+    return of(date, value, SKIP_WEEKENDS);
+  }
+
+  public static DenseLocalDateDoubleTimeSeries of(
+      LocalDate date,
+      double value,
+      DenseTimeSeriesCalculation dateCalculation) {
+    return new DenseLocalDateDoubleTimeSeries(date, new double[]{value}, dateCalculation);
+  }
+
+  /**
+   * Create a new time series with the specified additional points.
+   *
+   * @param additionalPoints  the points to add or update in the time series.
+   * @return a new time series containing updated points
+   */
+  // Should this not be here with the expectation that we always go through the builder?
```

Comment:
Update clearly can be efficient but is a bit weird. Can't be called update as that implies mutability. Maybe there is scope for `combineWith(DenseLDDTimeSeries)`.